ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -85,10 +89,13 @@
     this.clusterMap = clusterMap;
     responseHandler = new ResponseHandler(clusterMap);
     this.time = time;
-    ocList = new ArrayList<OperationController>(routerConfig.routerScalingUnitCount);
-    for (int i = 0; i < routerConfig.routerScalingUnitCount; i++) {
-      ocList.add(new OperationController(i));
+    ocCount = routerConfig.routerScalingUnitCount;
+    ocList = new ArrayList<>();
+    for (int i = 0; i < ocCount; i++) {
+      ocList.add(new OperationController(Integer.toString(i)));
     }
+    backgroundDeleter = new BackgroundDeleter();
+    ocList.add(backgroundDeleter);
```

Comment:
if the purpose of adding to `ocList` is only for shutdown, I would recommend against it. Since there is only instance of `BackgroundDeleter`, it should be easy to shutdown/wait for shutdown of that one. I feel that keeping the _real_ ocs separated from the utility ocs is useful - one e.g. are the changes required for `getOperationController()`.